lib/p2p/Pool.ts

```diff
@@ -183,6 +183,25 @@ class Pool extends EventEmitter {
    * Attempt to create an outbound connection to a node using its known listening addresses.
    */
   private tryConnectNode = async (node: NodeConnectionInfo, retryConnecting = false) => {
+    if (!await this.connectWithLastAddress(node)) {
+      await this.connectWithAdvertisedAddresses(node, retryConnecting);
```

Comment:
Let's stick to the `tryX` naming convention for functions which are swallowing errors.